### DISCUSSION TITLE
Make:all command

### DISCUSSION THREAD

**Discussion by dmdboi** [adonisjs/core#4632] [Ideas]
I believe a good enchancement to the v6 development experience would be to add a ``make:all`` command that could generate all the necessary files for a Resource. 

For example, run a single command and have a Model/Migration, Controller, Views and Routes generated. 

I've made an command similar to this for v5, but I believe v6 should have it baked in.
https://github.com/weskerjs/supergen

With a bit more work, the command could be adapted to take into account @layout() tags when generating view files too

**Comment by RomainLanz** [comment]
Hey @dmdboi! 👋🏻 

It's similar to what Rails does with their `scaffold` command.

```sh
bin/rails generate scaffold HighScore game:string score:integer
    invoke  active_record
    create    db/migrate/20190416145729_create_high_scores.rb
    create    app/models/high_score.rb
    invoke    test_unit
    create      test/models/high_score_test.rb
    create      test/fixtures/high_scores.yml
    invoke  resource_route
     route    resources :high_scores
    invoke  scaffold_controller
    create    app/controllers/high_scores_controller.rb
    invoke    erb
    create      app/views/high_scores
    create      app/views/high_scores/index.html.erb
    create      app/views/high_scores/edit.html.erb
    create      app/views/high_scores/show.html.erb
    create      app/views/high_scores/new.html.erb
    create      app/views/high_scores/_form.html.erb
    invoke    test_unit
    create      test/controllers/high_scores_controller_test.rb
    create      test/system/high_scores_test.rb
    invoke    helper
    create      app/helpers/high_scores_helper.rb
    invoke      test_unit
    invoke    jbuilder
    create      app/views/high_scores/index.json.jbuilder
    create      app/views/high_scores/show.json.jbuilder
    create      app/views/high_scores/_high_score.json.jbuilder
```

That could be doable, but I am unsure how you would handle the `@layout()` tags when generating a view. 😄

**Reply by dmdboi** [reply]
Absolutely! I took inspiration from their scaffold command 

I think if the user specified a layout via a --layout argument, the ejs templates would add the additional code (based on the supergene code)

**Reply by Bortolla** [reply]
Also, if the project is using Inertia with some other front-end framework (such as Vue) it can get a bit tricky - maybe just leave alone the generation of views?

Anyway, I agree that a Make:all command would be very useful, and give off a much more Rails like experience

**Reply by dmdboi** [reply]
I think we could have a --no-views parameter for that case and where people are using it in an API
